### ISSUE TITLE
RSPEC-3963 rule incompatible with readonly fields

### ISSUE THREAD

**Issue by metinogurlu** [SonarSource/sonar-dotnet#2421]
### Description

RSPEC-3963 "static" fields should be initialized inline rule is force to static readonly fields inline. But its impossible. Because "A static readonly field cannot be assigned to (except in a static constructor or a variable initializer"

### Expected behavior

readonly fields may be excluded from this rule

### Actual behavior

![actual](https://user-images.githubusercontent.com/17520202/57398909-d0b66600-71d8-11e9-94ff-89fef5e285b2.PNG)

**Comment by Corniel** [comment]
The whole point of this rule is to use a static constructor, and it is right in this case.

You should write your code like this:

``` C#
namespace SonarAnalyzer.Experiments.CSharp
{
    public static class S3963
    {
        public static readonly int MyConfigInt = Convert.ToInt32(true);
        public static readonly string MyConfigString = Convert.ToString("13");
    }
}
```
So you don't need a static constructor to assign your static readonly fields. That is what this rule is telling you.

Off-topic. It is a bad design to have a static class for your settings. This makes it (extremely) difficult to change the behavior when writing tests on it.
